### DISCUSSION TITLE
Code Files Mixed Line Endings

### DISCUSSION THREAD

**Discussion by SkyeHoefling** [dnnsoftware/Dnn.Platform#3995] [General] [answered]
Is there a recommended line endings in the DNN code base? Since we have added stylecop I am seeing just about every file I open up mixed line endings. Visual Studio is asking me to normalize this to resolve the errors. Do we have a strong stance on what we prefer such as `CRLF` which is the standard line ending on windows. 

We should define a strong opinion on this and add it to the `.editorconfig` this will ensure the line endings are corrected for every file that is opened in a supported IDE. **I might have missed something, but I didn't see this declared in either the editorconfig or stylecop settings.**

**Comment by bdukes** [comment, marked as answer]
It looks like the closest we have is CRLF specified in `.gitattributes` https://github.com/dnnsoftware/Dnn.Platform/blob/0bd8b22b997d953a0e59bd1d313314b0453d4e5f/.gitattributes#L2

There doesn't appear to be a StyleCop rule for line endings, but some of the auto-fixes may have introduced inconsistent line endings.

Let's add `crlf` to the `.editorconfig` :+1:

**Reply by SkyeHoefling** [reply]
Created #3997 to track work on this
